### PR TITLE
Vault integration: add CredentialLibrary resources

### PR DESCRIPTION
**Pull request by mgaffney**
Structs added:
* `vault.CredentialLibrary`
* `vault.Lease`

### CODE REVIEW

File: internal/credential/vault/vault_lease.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+}
+
+// SetTableName sets the table name.
+func (l *Lease) SetTableName(n string) {
+	l.tableName = n
+}
+
+func (l *Lease) oplog(op oplog.OpType) oplog.Metadata {
+	metadata := oplog.Metadata{
+		"resource-type": []string{"credential-vault-lease"},
```
Comment (louisruch) [inline internal/credential/vault/vault_lease.go:78]: Should you not have the `resource-public-id` added to the metadata?

**Review by louisruch — APPROVED** [review]
One question about the lease oplog metadata otherwise LGTM

File: internal/credential/vault/vault_lease.go (hunk 1 above)
Comment (mgaffney) [reply internal/credential/vault/vault_lease.go:78]: Which `resource-public-id`?

Comment (mgaffney) [reply internal/credential/vault/vault_lease.go:78]: Never mind. I see what you mean. Ty!

**Pull request merged**